_health-care/_js/reducers/veteran/index.js

```diff
@@ -187,17 +187,38 @@ const blankVeteran = {
 function veteran(state = blankVeteran, action) {
   let newState = undefined;
   switch (action.type) {
-    case VETERAN_FIELD_UPDATE:
+    case VETERAN_FIELD_UPDATE: {
       newState = Object.assign({}, state);
       _.set(newState, action.propertyPath, action.value);
       return newState;
+    }
 
-    case ENSURE_FIELDS_INITIALIZED:
+    case ENSURE_FIELDS_INITIALIZED: {
       newState = Object.assign({}, state);
       // TODO(awong): HACK! Assigning to the sub object assumes pathToData() returns a reference
       // to the actual substructre such that it can be reassigned to.
       Object.assign(pathToData(newState, action.path), initializeNullValues(pathToData(newState, action.path)));
       return newState;
+    }
+
+    // Copies the veteran's address into the spouse's address fields if they have the same address.
+    // Clears the spouse's address fields if they do not.
+    case UPDATE_SPOUSE_ADDRESS: {
+      newState = Object.assign({}, state);
+      const emptyAddress = {
+        street: null,
+        city: null,
+        country: null,
+        state: null,
+        zipcode: null,
+      };
+      if (action.value) {
+        _.set(newState, action.propertyPath, state.veteranAddress.address);
+      } else {
```

Comment:
Do we need to set this to empty the spouse's address fields if they don't have the same address? Wouldn't `spouseAddress` just update as normal if they didn't have the same address?